src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerGenotypingEngine.java

```diff
@@ -168,24 +273,82 @@ public CalledHaplotypes(final List<VariantContext> calls, final Set<Haplotype> c
+                        }
+                    } else if (spanningEvent.getReference().length() < mergedVC.getReference().length()) {
+                        // spanning event has shorter ref allele than merged VC; we need to pad out its alt allele
+                        final Map<Allele, Allele> spanningEventAlleleMappingToMergedVc
+                                = GATKVariantContextUtils.createAlleleMapping(mergedVC.getReference(), spanningEvent, new ArrayList<>());
+                        final Allele remappedSpanningEventAltAllele = spanningEventAlleleMappingToMergedVc.get(spanningEvent.getAlternateAllele(0));
+                        // in the case of GGA mode the spanning event might not match an allele in the mergedVC
+                        if (result.containsKey(remappedSpanningEventAltAllele)) {
+                            result.get(remappedSpanningEventAltAllele).add(h);
+                        }
+                    } else {
+                        // the process of creating the merged VC in AssemblyBasedCallerUtils::makeMergedVariantContext should have
+                        // already padded out the reference allele, therefore this spanning VC must not be in events at this site
+                        // because we're in GGA mode and it's not an allele we want
+                        continue;
+                    }
+
+                } else {
+                    // the event starts prior to the current location, so it's a spanning deletion
+                    if (activeAllelesToGenotype != null && activeAllelesToGenotype.size() > 0) {
+                        // in HC GGA mode we need to check to make sure that spanning deletion
+                        // events actually match one of the alleles we were given to genotype
+                        VariantContext matchingGivenVc = findMatchingGivenAllele(activeAllelesToGenotype, spanningEvent);
+                        if (matchingGivenVc != null) {
+                            if (!result.containsKey(Allele.SPAN_DEL)) {
+                                result.put(Allele.SPAN_DEL, new ArrayList<>());
+                            }
+                            result.get(Allele.SPAN_DEL).add(h);
+                        }
+
+                    } else {
+                        if (! result.containsKey(Allele.SPAN_DEL)) {
+                            result.put(Allele.SPAN_DEL, new ArrayList<>());
+                        }
+                        result.get(Allele.SPAN_DEL).add(h);
                     }
+                    break;
                 }
             }
+
+            if (! hasSpanningEvents) {    //no events --> this haplotype supports the reference at this locus
+                result.get(ref).add(h);
+            }
         }
         return result;
     }
 
+    private static VariantContext findMatchingGivenAllele(final List<VariantContext> activeAllelesToGenotype, final VariantContext spanningEvent) {
```

Comment:
Couldn't this conceivably return `List<VariantContext>` in the case of deletions spanning multiallelic GGA vcs?  (Or maybe, multiple GGA vcfs with different alt alleles at the same site).

Reply:
After looking at this again, changed it to a boolean method that returns true if it finds a matching given allele, since that's all it was being used for anyway.